src/def_tools.jl

```diff
@@ -0,0 +1,77 @@
+# These utilities are for working with the signature_def `Dict` that comes out of
+# `signature`/`splitdef`
+
+
+
+"""
+    args_tuple_expr(signature_def::Dict{Symbol})
+    args_tuple_expr(arg_exprs)
+
+For `arg_exprs` being a list of arguments expressions from a signature, of a form
+such as `[:(x::Int), :(y::Float64), :(z::Vararg)]`, or being a whole signature_def `Dict`
+containing a `signature_def[:args]` value of that form.
+
+This returns a tuple expresion containing all of the args by name; while correctly handling
+splatting, for things that are `Vararg` typed. e.g for the prior example `:((x, y, z...))`
+
+This is useful for modifiying the `signature_def[:body]`.
+For example, one could printout all the arguments via
+
+```julia
+signature_def[:body] = quote
+    args = \$(args_tuple_expr(signature_def))
+    println("args = ",args)
+    \$(signature_def[:body]) # insert old body
+end
+```
+
+A more realistic use case is if you want to insert a call to a another function
+that accepts same argument as the original function.
+"""
+function args_tuple_expr end
+
+args_tuple_expr(signature_def::Dict{Symbol}) = args_tuple_expr(signature_def[:args])
+
+function args_tuple_expr(arg_exprs)
+    ret = Expr(:tuple)
+    ret.args = map(arg_exprs) do arg
+        arg isa Symbol && return arg  # it is just `x` not `x::T`
```

Comment:
can we just delete this since it is handled below? or do we want to short circuit here?

Reply:
Oh, yeah, i forgot to delete this, yes we can.
I added features incrementally and minimally, one test at a time.

```suggestion
```